content/en/docs/pipelines/tutorials/benchmark-examples.md

```diff
@@ -0,0 +1,184 @@
+times simultaneously on a given Kubeflow Pipelines deployment, and collects the
+server latencies and run durations. Among all the operations that the Kubeflow
+Pipelines can perform, running a pipeline is arguably the most unpredictable and
+costly one. Other operations, e.g., creating a pipeline (version) or creating an
+experiment, usually induce a predicatable and moderate cost. E.g., creating a
+pipeline version will introduce a new row in the pipeline versions table, a new
+file in minio server. The new file's size depends on the pipeline version's
+manifest. If we exclude the rare case of an extremely large manifest and assume
+an average sized manifest for each created pipeline version, the total cost of
+creating a pipeline version grows linearly with the number of pipeline versions.
+However, on the other hand, the cost of running a pipeline or a pipeline version
+involves much more uncertainty and sometimes quite high a cost. A pipeline or a
+pipeline version can have arbitrary components and hence running a pipeline or a
+pipeline version can incur arbitrary time and space complexities. E.g., the step
+in a pipeline can use a customized container imange which performs a super
+expensive training task. In addition, the runs in a Kubeflow Pipelines deployment
+also consume more DB space than pipelines, pipeline versions, experiments, etc.
+
+Therefore, when the performance and scalibility of Kubeflow Pipelines are in
+question, the simulated benchmark workloads are intended to focus on the
+operation of running a pipeline or a pipeline version in order to targe the pain
+point of Kubeflow Pipelines and reveal more useful information.
+
+## Prerequisites for Running Benchmark Scripts
+
+To use the provided benchmark scripts, you will need:
+
+1. A Jupyter notebook instance
+1. A Kubeflow Pipelines deployment
+1. A benchmark script
+1. A pipeline manifest
+
+The Jupyter notebook instance and the Kubeflow Pipelines deployment needed for
+running a benchmark can be hosted either on cloud or on a local machine, as long
+as the Jupyter notebook instance has proper accesses to the Kubeflow Pipelines
+deployment, e.g., be able to CREAT, GET, DELETE and LIST the pipeline, pipeline
+version, run, job and experiment in the Kubeflow Pipelines deployment.
+
+One way of setting up everything and runnning a benchmark script is shown below
+as an example.
+
+## An Example of Running Benchmark Scripts
+
+In this example, a Kubeflow Pipelines deployment is hosted on cloud and a
+Jupyter notebook instance is hosted on a local machine. The cloud cluster where
+the Kubeflow Pipelines deployment resides have two node pools: one is for
+running the Kubeflow Pipelines servers, e.g., ml-pipeline and ml-pipeline-ui,
+and the other is for running the pipelines.
+
+Moreover, in this example, the benchmark script [run_service_api.ipynb](https://github.com/jingzhang36/pipelines/blob/different_tools/tools/benchmarks/run_service_api.ipynb)
```

Comment:
I can submit the notebook file .ipynb to kubeflow/pipelines repo. But for the pipeline manifest yaml file, putting it on github and use the github link instead of the storage.googleapis.com would cause problem. The github link of the yaml file, when used with api method, doesn't always work and usually we use storage.googleapis.com to avoid possible issues.